crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -1000,6 +1003,29 @@ impl CumulocityConverter {
         Ok(registration_messages)
     }
 
+    fn validate_operation_supported(
+        &self,
+        op_type: &OperationType,
+        topic_id: &EntityTopicId,
+    ) -> Result<(), ConversionError> {
+        let target = self.entity_store.try_get(topic_id)?;
+
+        match target.r#type {
+            EntityType::MainDevice => Ok(()),
+            EntityType::ChildDevice => match &target.parent {
+                Some(parent) if !parent.is_default_main_device() => {
+                    Err(ConversionError::UnexpectedError(anyhow!(
```

Comment:
It's really okayish to say `UnexpectedError` only because it's for this kind of quick temporary fix. However, this error is not a true unexpected error. Creating an operation for nested child devices is defenitetly a standard usecase. Just thin-edge.io is not ready for that yet. For me, it should be UnsupportedFeatureError or whatever.

Reply:
Yeah, I didn't introduce a new variant for that same reason (add something new that will be removed very soon).